Fix #60494: query doesn't work on DataFrame integer column names

David-msggc opened the pull request:
- [x] closes #60494
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.
Some io code checks failed but they were already failing before the bugfix.
Function _get_cleaned_column_resolvers ignores integer column names, so when it is called on eval function, it returns empty columns which it shouldn't since there is an integer column. Converting the integer columns to strings before calling the _get_cleaned_column_resolvers on eval fucntion fixes this.

David-msggc committed:
Fix #60494: query doesn't work on DataFrame integer column names

mroeschke commented on pandas/core/frame.py:
```diff
@@ -4790,6 +4790,12 @@ def eval(self, expr: str, *, inplace: bool = False, **kwargs) -> Any | None:
+
+        if any(isinstance(col, int) for col in self.columns):
+            self = self.rename(
+                columns={col: f"{col}" for col in self.columns if isinstance(col, int)}
```
This should probably be handled in `clean_column_name`

David-msggc replied:
But if I handle it in clean_column_name instead of in eval, wouldn't it return the same error since the integer column names would be ignored because of the line [ if not isinstance(k, int) ] in _get_cleaned_column_resolvers, thus not reaching the clean_column_name function. Maybe it can be handled in _get_cleaned_column_resolvers ?

mroeschke replied:
Maybe the `if not isinstance(k, int)` needs to be removed in `_get_cleaned_column_resolvers` instead. I would see what breaks if you remove that.

David-msggc committed:
fix: removed line that ignores an integer column name from _get_cleaned_column_resolvers

Also removed the changes from the previous commit, which converted integer column names to strings

mroeschke commented:
Thanks @David-msggc
